coderd/database/migrations/000178_provisioner_daemon_idx_owner.up.sql

```diff
@@ -0,0 +1,10 @@
+ALTER TABLE ONLY provisioner_daemons
+	DROP CONSTRAINT IF EXISTS provisioner_daemons_name_key;
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_provisioner_daemons_name_owner_key
```

Comment:
Out of curiosity, do we need an index since Day 1? What was the upper limit of provisioner daemons we hit in scaletests?

Reply:
We need to at least keep the existing unique index on name.
However, this will allow multiple user-level provisioners to be named "my-provisioner".
Otherwise, each user will have to choose a unique provisioner name.

Reply:
Gotcha, I ignored the use case with multiple users 👍  Thanks